optuna/storages/in_memory.py

```diff
@@ -434,12 +434,12 @@ def _check_study_id(self, study_id):
         # type: (int) -> None
 
         if study_id not in self._studies:
-            raise ValueError("No study with study_id {} exists.".format(study_id))
+            raise KeyError("No study with study_id {} exists.".format(study_id))
 
     def _check_trial_id(self, trial_id: int) -> None:
 
         if trial_id not in self._trial_id_to_study_id_and_number:
-            raise ValueError("No trial with trial_id {} exists.".format(trial_id))
+            raise KeyError("No trial with trial_id {} exists.".format(trial_id))
```

Comment:
[Comment] The `trial_id` is an internal attribute, but I think it is reasonable to use it here because this `KeyError` is not usually raised by user-code problems.